rules/jsll/src/rules/jsll-awa-init/jsll-awa-init.ts

```diff
@@ -23,89 +23,114 @@ const debug: debug.IDebugger = d(__filename);
 const rule: IRuleBuilder = {
     create(context: RuleContext): IRule {
         const linter = new Linter();
-        let validated: boolean = false; // If `validateAwaInit` has run.
-        let hasJSLLScript: boolean = false; // If link to JSLL scripts has been included.
+        let isHead: boolean = true;
+        let validated: boolean = false; // If the linter has run.
+        let immediateAfterJSLLscript: boolean = false; // If JSLL init should be validated.
+        let traverseStarts: boolean = false;
+        let tempInit: IScriptParse | null = null; // A temporary cache for JSLL init script content.
 
         linter.defineRule('jsll-awa-init', {
             create(eslintContext) {
                 let isFirstExpressionStatement: boolean = true;
 
                 return {
                     ExpressionStatement(node) {
-                        if (!isFirstExpressionStatement) { // Only the first expression statement should be checked.
-                            return;
-                        }
+                        const expression = node.expression;
+                        const isInit = isInitCode(expression);
 
-                        isFirstExpressionStatement = false;
+                        if (isInit) {
+                            if (!isFirstExpressionStatement) {
+                                eslintContext.report(node, '"awa.init" is not called as soon as possible.');
+                            }
 
-                        validateAwaInit(node, eslintContext, true);
+                            if (!configIsDefined(node, eslintContext)) {
+                                eslintContext.report(node, `The variable passed to "awa.init" is not defined.`);
+                            }
+                        }
 
+                        isFirstExpressionStatement = false;
```

Comment:
@sarvaje This assignment has to wait after the following code because `isFirstExpressionStatement` is used in it: 
```
                        if (isInit) {
                            if (!isFirstExpressionStatement) {
                                eslintContext.report(node, '"awa.init" is not called as soon as possible.');
                            }

                            ...
                        }
```

Reply:
Maybe you should use another variable instead. Because the code run asynchronously, if the rule `jsll-awa-init` is called two times at the same time, it would be possible to run two times the same code, and report two times the same error.